src/main/java/io/github/vcuswimlab/stackintheflow/model/L2H/L2HPredictor.java

```diff
@@ -0,0 +1,182 @@
+			return Double.compare(other.probability, probability);
+		}
+	}
+
+	private Map<String, Integer> nameToIdMap;
+	private CustomL2H testSampler;
+	private LabelTextDataset data;
+
+	public L2HPredictor() throws IOException {
+		nameToIdMap = new HashMap<>();
+		BufferedReader br = new BufferedReader(new FileReader(formatFolder+formatFile+".wvoc"));
+		int maxId = -1;
+		try {
+			String line;
+			int intId = 0;
+			while ((line = br.readLine()) != null) {
+				nameToIdMap.put(line, intId);
+				intId++;
+			}
+		} finally {
+			br.close();
+		}
+
+		data = new LabelTextDataset(datasetName);
+        data.setFormatFilename(formatFile);
+        data.loadFormattedData(formatFolder);
+        data.filterLabelsByFrequency(minLabelFreq);
+		data.prepareTopicCoherence(numTopWords);
+
+		testSampler = new CustomL2H();
+		testSampler.setVerbose(true);
+        testSampler.setDebug(false);
+        testSampler.setLog(false);
+        testSampler.setReport(false);
+        testSampler.configure("./output", data.getWordVocab().size(), alpha, beta, a0, b0,
+				new MSTBuilder(data.getLabels(), data.getLabelVocab()), false, false, AbstractSampler.InitialState.PRESET, PathAssumption.MAXIMAL, false, burnIn, modelIterations, sampleLag, 1);
+        testSampler.setTestConfigurations(burnIn, modelIterations, sampleLag);
+	}
+
+	public void computeL2HPredictions(String fileToPredict, String outputFileName, String tempFileName) throws Exception {
+		BufferedReader br;
+		br = new BufferedReader(new FileReader(fileToPredict));
+		Map<Integer, Integer> wordOccurrenceMap = new HashMap<>();
+		try {
+			String line;
+			while ((line = br.readLine()) != null) {
+				String[] parts = line.split("\\s+");
+				for(int i = 1; i < parts.length; i++) {
+					String name = parts[i-1] + " " + parts[i];
+					if(nameToIdMap.containsKey(name)) {
+						int index = nameToIdMap.get(name);
+						if(wordOccurrenceMap.containsKey(index)) {
+							wordOccurrenceMap.put(index, wordOccurrenceMap.get(index)+1);
+						} else {
+							wordOccurrenceMap.put(index, 1);
+						}
+					}
+				}
+			}
+		} finally {
+			br.close();
+		}
+
+		BufferedWriter bw = new BufferedWriter(new FileWriter(tempFileName));
+		try {
+			bw.write(wordOccurrenceMap.size() + " ");
+			for(Map.Entry<Integer,Integer> e : wordOccurrenceMap.entrySet()) {
+				bw.write(e.getKey() + ":" + e.getValue() + " ");
+			}
+		} finally {
+			bw.close();
+		}
+		int[][] newWords = new TextDataLoader("dummyDataset",tempFileName).getWords();
+		double[][] initPredictions = computeInitPredictions(newWords, data.getLabelVocab().size());
+
+		try {
+            testSampler.sampleNewDocuments(stateFile, newWords, outputFileName, initPredictions, 20);
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new RuntimeException();
+        }
+	}
+
+	public void computeMostLikelyTags(String predictionsFileName, List<String> tagNamesOut, List<Double> tagProbabilitiesOut, int topN) throws Exception {
```

Comment:
See above, I think it would be better to return an object containing the  tagNamesOut and tagProbabilitiesOut collections, rather than passing them in to be modified.

Reply:
Also, would it make more sense to return a sorted map of the tag name to the tag probability, that way they can easily be matched to one another?

Reply:
Agreed, computeL2HPredictions now returns a collection of a TagPrediction class that solves these issues.